daemons/controld/controld_execd.c

```diff
@@ -1765,48 +1765,52 @@ do_lrm_invoke(long long action,
     user_name = pcmk__update_acl_user(input->msg, F_CRM_USER, NULL);
     crm_op = crm_element_value(input->msg, F_CRM_TASK);
     from_sys = crm_element_value(input->msg, F_CRM_SYS_FROM);
-    if (!pcmk__str_eq(from_sys, CRM_SYSTEM_TENGINE, pcmk__str_casei)) {
+    if (!pcmk__str_eq(from_sys, CRM_SYSTEM_TENGINE, pcmk__str_none)) {
         from_host = crm_element_value(input->msg, F_CRM_HOST_FROM);
     }
-    crm_trace("Executor %s command from %s as user %s",
-              crm_op, from_sys, user_name);
 
-    if (pcmk__str_eq(crm_op, CRM_OP_LRM_DELETE, pcmk__str_casei)) {
-        if (!pcmk__str_eq(from_sys, CRM_SYSTEM_TENGINE, pcmk__str_casei)) {
+    if (pcmk__str_eq(crm_op, CRM_OP_LRM_DELETE, pcmk__str_none)) {
+        if (!pcmk__str_eq(from_sys, CRM_SYSTEM_TENGINE, pcmk__str_none)) {
             crm_rsc_delete = TRUE; // from crm_resource
         }
         operation = CRMD_ACTION_DELETE;
 
-    } else if (pcmk__str_eq(crm_op, CRM_OP_LRM_FAIL, pcmk__str_casei)) {
-        fail_lrm_resource(input->xml, lrm_state, user_name, from_host,
-                          from_sys);
-        return;
-
     } else if (input->xml != NULL) {
         operation = crm_element_value(input->xml, XML_LRM_ATTR_TASK);
     }
 
-    if (pcmk__str_eq(crm_op, CRM_OP_LRM_REFRESH, pcmk__str_casei)) {
+    CRM_CHECK(!pcmk__str_empty(crm_op) || !pcmk__str_empty(operation), return);
+
+    crm_trace("'%s' execution request from %s as %s user",
+              pcmk__s(crm_op, operation),
+              pcmk__s(from_sys, "unknown subsystem"),
+              pcmk__s(user_name, "current"));
+
+    if (pcmk__str_eq(crm_op, CRM_OP_LRM_FAIL, pcmk__str_none)) {
+        fail_lrm_resource(input->xml, lrm_state, user_name, from_host,
+                          from_sys);
+
```

Comment:
Don't we come here just if we want the executor to explicitly fail a resource.
Is it intentional then to fall through to the error-message below?
Especially unknown command seems a bit misleading.